src/BL/Commands/FindMaxTemperatureCommand.cs

```diff
@@ -0,0 +1,98 @@
+﻿using Exadel.Forecast.BL.Interfaces;
+using Exadel.Forecast.DAL.Models;
+using Exadel.Forecast.Models.Interfaces;
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Exadel.Forecast.BL.Commands
+{
+    public class FindMaxTemperatureCommand : ICommand
+    {
+        private readonly IConfiguration _configuration;
+        private readonly IValidator<string> _cityValidator;
+        private readonly IValidator<double> _temperatureValidator;
+        private readonly string _cityNames;
+
+        public FindMaxTemperatureCommand
+            (
+                string cityNames,
+                IConfiguration configuration,
+                IValidator<string> cityValidator,
+                IValidator<double> temperatureValidator
+            )
+        {
+            _configuration = configuration;
+            _cityValidator = cityValidator;
+            _temperatureValidator = temperatureValidator;
+            _cityNames = cityNames;
+        }
+
+        public async Task<string> GetResultAsync()
+        {
+            if (!_cityValidator.IsValid(_cityNames))
+            {
+                return "You need to type name of the city!";
+            }
+
+            var forecastRepository = _configuration.GetDefaultForecastApi();
+            string[] cityNames = _cityNames.Split('\u002C').Select(p => p.Trim()).ToArray();
+            Task<CurrentResponseModel>[] tasks = new Task<CurrentResponseModel>[cityNames.Length];
+            Stopwatch stopwatchAll = new Stopwatch();
+            stopwatchAll.Start();
+            for (int i = 0; i < cityNames.Length; i++)
+            {
+                tasks[i] = forecastRepository.GetTempByNameAsync(cityNames[i]);
+            }
+            CurrentResponseModel[] responses = await Task.WhenAll(tasks);
+            stopwatchAll.Stop();
```

Comment:
You control execution time for all cities but in the task you have "City name and how long the request took in ms. "